runtime/doc/api.txt

```diff
@@ -2400,25 +2402,27 @@ nvim_buf_set_text({buffer}, {start_row}, {start_col}, {end_row}, {end_col},
                   {replacement})
                 Sets (replaces) a range in the buffer
 
-                This is recommended over nvim_buf_set_lines when only
+                This is recommended over |nvim_buf_set_lines()| when only
                 modifying parts of a line, as extmarks will be preserved on
                 non-modified parts of the touched lines.
 
-                Indexing is zero-based and end-exclusive.
+                Row indices are zero-based and end-inclusive. Column indices
+                are zero-based, end-exclusive, and correspond to bytes, not
+                character columns (see |col()| vs. |charcol()|).
```

Comment:
hmm, is `charcol()` something that is used very often? Typically we like to emphasize the distinction between (byte) column vs _virtual column_ which also takes double-width into account. These two are the main concepts that are often used in the API, functions and commands, and interaction with the user, and introducing more intermediates (offset with composing or uncomposed chars) is likely not helpful at this point.